functors task (1)/student/include/impl.h

```diff
@@ -1,13 +1,27 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <vector>
+#include <map>
+
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()()
+    {
+        if(mCurrentBrand == OrderedBrands.end())
+        {
+            mCurrentBrand = OrderedBrands.begin();
+        }
+        return *(mCurrentBrand++);
+    }
+private:
+    static const std::vector<BeerBrand> OrderedBrands;
```

Comment:
You're right, but here we have guarantee about the enum values order (it won't be changed).
This is real-world solution. Task solution is simpler.
But it is still accepted!